### PR TITLE
Refactor/generatebudget

### PR DESCRIPTION
**Pull request by alysonjan**

### CODE REVIEW

File: routes/api/dean/GenerateSubject.js [hunk 1]
```diff
@@ -37,8 +50,12 @@ router.post('/add', [auth,
...
                     } else {
                         result.forEach(subjectBudget => {
-                            const sqlInsertGenerateSubject = "INSERT INTO generate_subject (college_name,curriculum_code,subject_code,semester,year_level,subject_name,lec_units,lab_units,total_units,lec_teaching_hours,lab_teaching_hours,total_teaching_hours,students,no_of_sections,no_of_faculty,rad_hours) VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)";
-                            db.query(sqlInsertGenerateSubject, [subjectBudget.collegeName, subjectBudget.programName, subjectBudget.courseCode, subjectBudget.sem, subjectBudget.yearLevel, subjectBudget.subjectName, subjectBudget.lecUnits, subjectBudget.labUnits, subjectBudget.totalUnits, subjectBudget.lecTeachingHours, subjectBudget.labTeachingHours, subjectBudget.totalTeachingHours, students, no_of_sections, total_teaching_hours, no_of_faculty], (err, result1) => {                      })
+                            let overall_teaching_hours = no_of_sections*subjectBudget.totalTeachingHours;
+                            let no_of_faculty = overall_teaching_hours/24;
```
Comment (HiroAcocoro) [inline routes/api/dean/GenerateSubject.js]: can't be stored in constants?

File: routes/api/dean/GenerateSubject.js [hunk 2]
```diff
@@ -37,8 +50,12 @@ router.post('/add', [auth,
...
+                            let rad = 70;
```
Comment (HiroAcocoro) [inline routes/api/dean/GenerateSubject.js]: can't be stored in constants?

File: routes/api/queries/Query.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+const express = require('express');
+const router = express.Router();
+// const { check, validationResult } = require('express-validator');
```
Comment (HiroAcocoro) [inline routes/api/queries/Query.js]: residual comment?

File: routes/api/queries/Query.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+const db = require('../../../config/database');
+const auth = require('../../../middleware/auth');
+
+router.get('/curriculum', auth, async(req,res)=>{
+    try {
+        const sqlSelectCurriculum = "SELECT DISTINCT curriculum_code,year_level FROM curriculum";
+        db.query(sqlSelectCurriculum, (err, result) =>{
+            if (err){
+                res.status(400).json({ errors: [{ msg: 'Oops, Something went wrong'}] });
```
Comment (HiroAcocoro) [inline routes/api/queries/Query.js:11]: Generic server error, store in constants>Messages.js>Errors

File: routes/api/dean/GenerateSubject.js (hunk 1 above)
Comment (alysonjan) [reply routes/api/dean/GenerateSubject.js]: oh i see, the variable 24 is that what u mean? ok i will boss

File: routes/api/dean/GenerateSubject.js (hunk 2 above)
Comment (alysonjan) [reply routes/api/dean/GenerateSubject.js]: my bad, I forgot to put it on constants sorry

**Pull request merged**